src/main/java/duke/Task.java

```diff
@@ -0,0 +1,86 @@
+package duke;
+import java.time.LocalDate;
+
+/**
+ * Task class which stores information of the task that users input to the chatbot.
+ */
+public class Task {
+
+    protected String description;
+    protected boolean isDone;
+    protected String type;
+    protected LocalDate by;
+
+    /**
+     * Constructs a Task object.
+     *
+     * @param description   Description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructs a Task object.
+     *
+     * @param description   Description of the task.
+     * @param by    Date of when the task is due by/at.
+     */
+    public Task(String description, String by) {
+        this.description = description;
+        this.by = LocalDate.parse(by);
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a String to show if a task is done.
+     * If the task is done, it will return X.
+     * Else, it will return a whitespace.
+     *
+     * @return  X or whitespace depending on whether the task is done.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Returns a String of the description of the task.
+     *
+     * @return  description of the task
+     */
+    public String getDescription() {
+        return this.description;
+    }
```

Comment:
```suggestion
    public String getDescription() {
        return description;
    }
```
Is it necessary to include `this` here? (since removing it does not make the code ambiguous)